I3uckwheat/cards-of-carousal
# Feature/116 connect player join screen to host

### Description

These changes expand on the groundwork previously done in the app in order to implement the connection between Host and Player through PlayerJoinScreen. A JOIN_LOBBY event is dispatched as the user clicks the _Let's Carouse_ button, sending the input information of Player Name and Join Code through the socket to the backend and then from the backend it is forwarded to the Host. From this user interaction, it is first established the socket connection that will be used to exchange messages back and forth between the player and the host.

In order to achieve this, a few modifications in the backend `Lobby` and `LobbyList` classes were made, specifically to accommodate the player name in the functions `addPlayer` and `joinLobby` as you may see here.

These changes closes #116 

### Screenshots
#### Host pregame screen showing connected players
![image](https://user-images.githubusercontent.com/22618438/111999907-57e2fd80-8b15-11eb-958f-9c0afad42fd2.png)

#### Player's screen after joining the lobby
![image](https://user-images.githubusercontent.com/22618438/111999963-69c4a080-8b15-11eb-9880-3e736e15042f.png)


### How to Test
 
1. Pull the branch locally with `git switch feature/116-connect-player-join-screen-to-host`
2. Start the backend portion of the app with `npm run dev:backend`

3. In a new terminal window, start the frontend with `npm run dev:frontend`

4. In the browser, open up a tab @ localhost:3000
    4.1. Click the HOST button. You'll be redirected to the Host Pregame Screen.
    4.2. Copy the Join Code in the lower left panel

5. Open up another tab @ localhost:3000
    5.1. Click the JOIN button. You'll be redirected to the Player Join Screen.
    5.2. Fill in the inputs with your name and the Join Code you got from the Host Pregame Screen
    5.3. Click the LET'S CAROUSE button. You'll be redirected to the Player Message Screen, in which you should see a confirmation that you have successfully joined the lobby and are waiting for the host to start the game.
    5.4? If the Join Code is incorrect, you should see an indefinite "Connecting to lobby, please wait" message instead. 

6.  Check now the Host browser tab to see the player in the lobby.

7. You may also open multiple other tabs to connect multiple players to the lobby.
